### PR TITLE
refactor: Common static analysis issues

### PR DESCRIPTION
**Pull request by knutwannheden**
Use this link to re-run the recipe: https://app.moderne.io/recipes/org.openrewrite.staticanalysis.CommonStaticAnalysis?organizationId=T3BlblJld3JpdGU%3D

### CODE REVIEW

File: src/main/java/org/openrewrite/analysis/InvocationMatcher.java [hunk 1]
```diff
@@ -107,10 +107,8 @@ public boolean isSelect(Cursor cursor) {
         public boolean isAnyArgument(Cursor cursor) {
             return asExpression(
                     cursor,
-                    expression -> nearestMethodCall(cursor).map(call -> {
-                        return call.getArguments().contains(expression)
-                               && matcher.matches(call); // Do the matcher.matches(...) last as this can be expensive
```
Comment (JLLeitschuh) [inline src/main/java/org/openrewrite/analysis/InvocationMatcher.java:112]: Why have you removed this comment? This was important.

**Review by JLLeitschuh — COMMENTED** [review]
Some concerns

**Comment by knutwannheden** [comment]
@JLLeitschuh Thanks for your review! I ran a recipe which did all these changes. The switch-to-if is due to https://docs.openrewrite.org/recipes/staticanalysis/minimumswitchcases (RSPEC-1301). Not sure I agree with that particular one either...

The lost comments are deficiencies of the respective recipes. I can try to fix these.

**Comment by JLLeitschuh** [comment]
> The lost comments are deficiencies of the respective recipes. I can try to fix these.

Much appreciated. Can you also restore the lost comments in the meantime?

**Comment by knutwannheden** [comment]
> Much appreciated. Can you also restore the lost comments in the meantime?

https://github.com/openrewrite/rewrite-analysis/commit/112b76e14c2401d078f18f35b83d1ba4fcc19ff0

**Pull request merged**